native/explorer/src/encoding.rs

```diff
@@ -354,6 +387,7 @@ pub fn list_from_series(data: ExSeries, env: Env) -> Result<Term, ExplorerError>
         DataType::Binary => {
             generic_binary_series_to_list(&data.resource, s.binary()?.downcast_iter(), env)
         }
+        DataType::Categorical(Some(mapping)) => categorical_series_to_list(s, env, mapping),
```

Comment:
This is tricky. I wonder if we should return the categories (as strings) or their indexes in the mapping. And I think `term_from_value` should return the string.

`iovec_from_series` should definitely return indexes though.

Reply:
We don't want to avoid UInt32. But we want to avoid multiple representations for the same dtype. I.e. integers should have only a single representation. Therefore it is fine to return UInt32 for categorical types. :)